CMakeLists.txt

```diff
@@ -92,10 +92,13 @@ if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
   set(CMAKE_BUILD_TYPE "Release" CACHE STRING "Choose the type of build." FORCE)
 endif()
 
+if(NOT CMAKE_CXX_STANDARD)
 # Force C++11 (No extensions) and C99
 set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED True)
 set(CMAKE_CXX_EXTENSIONS OFF)
+endif()
+
```

Comment:
This is typical for any project and it's not hurting since by default it will use c++11

But if someone uses `-D CMAKE_CXX_STANDARD=14` (which is something you use for many projects) or any other than 11, we do not use this input and not give a warning that we disregard the flag. I think we should leave it.

Reply:
I really dislike that we need to hardcode 17 in the cmake of all Kokkos files but for now we need this for your CI. As soon as Kokkos move to another standard I will immediately remove this and we will have to update the CI to support any version. When I compile on Summit and I set the standard by hand hardcoding is not required.

Reply:
> This is typical for any project

That is not either typical or a _recommended_ setting for every Cmake project, some projects might be built using different CXX standards, that is not our case project-wide in our current ADIOS2.

> it's not hurting since by default it will use c++11

I do not see the need of this since ADIOS2 is not using any C++17/20 feature and ADIOS2 needs C++11 as min c++ revision; It is not in the scope of this PR to have this change; This might allow other users/package management software to build adios with other CXX_STANDARD and trigger strange errors.

Lets keep it simple, we do not need this really. If it comes the time we can add this feature.

Reply:
On another note if you need some files to be build with C++17 such as `AdiosKokkos.cpp` you can set the property CXX_STANDARD=17 for the `adios_core_kokkos` library.

Reply:
I'll remove this from here because the workaround of hardcoding 17 makes this irrelevant for this PR but I will open an issue and make another PR with this because I think it's unacceptable to not be able to set the standard. 
When the users/package management software specifies a standard they mean to use. It's unacceptable to ignore it and force C++11. I don't know of any project that does this (I looked even in cmake itself)

Reply:
For instance, VTK/ParaView/VTK-m they all are only compatible with a single standard same as ADIOS2, sure it will be nice to support different standards but we are not there.

>  I will open an issue and make another PR with this

That is a good idea